compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/classpathDiff/ClasspathSnapshotShrinker.kt

```diff
@@ -174,27 +174,36 @@ private fun ClasspathSnapshot.getNonDuplicateClassSnapshots(): List<ClassSnapsho
 
 /** Used by [shrinkAndSaveClasspathSnapshot]. */
 private sealed class ShrinkMode {
-    object NoChanges : ShrinkMode()
+    object UnchangedLookupsUnchangedClasspath : ShrinkMode()
 
-    class IncrementalNoNewLookups(
-        val shrunkCurrentClasspathAgainstPreviousLookups: List<AccessibleClassSnapshot>,
+    class UnchangedLookupsChangedClasspath(
```

Comment:
Have you considered simplifying this type hierarchy and just having NonIncremental one and the one that specifies lookupSymbols/currentClasspath/shrunkCurrent?

Reply:
The main algorithm below handles the 4 cases differently with no shared logic:
```
1. UnchangedLookupsUnchangedClasspath
2. UnchangedLookupsChangedClasspath
3. ChangedLookups
4. NonIncremental
```    

Case 3 again has 2 sub-cases which we also handle differently:
```
3a. ChangedLookupsUnchangedClasspath
3b. ChangedLookupsChangedClasspath
```

I can't group them because they have no shared logic (and removing this `ShrinkMode` class entirely would make the code a bit hard to read).

Reply:
If it is not possible to simplify, let's keep it as-is. Thanks for looking into it.